operator/pkg/rest/getter.go

```diff
@@ -51,9 +53,12 @@ func (c *ManifestRESTClientGetter) ToDiscoveryClient() (discovery.CachedDiscover
 		return nil, err
 	}
 
-	c.client = memory.NewMemCacheClient(discoveryClient)
+	memCachedClient := memory.NewMemCacheClient(discoveryClient)
+	if c.kymaObjKey.Name != "" {
```

Comment:
this access is not concurrency safe

Reply:
The previous implementation was blocking for all operations until mutex was available. This has been changed with the latest commit.

Reply:
Current implementation is blocking in the sync.Map for concurrent read, so same effect now?

Reply:
I know. with the latest commit, I meant the upcoming [commit](https://github.com/kyma-project/module-manager/pull/145/commits/b0addec544f4c1bc19c99d9f5ca69842746f96da) :)